packages/terra-wallet-provider/lib/TerraWalletProvider.ts

```diff
@@ -159,16 +174,18 @@ export default class TerraWalletProvider extends WalletProvider {
       txData = {
         ...(fee && {
           gasPrices: new Coins({
-            [this._network.asset]: fee
+            [this._asset]: fee
           })
         })
       }
     } else {
+      const feeAsset = this._tokenAddress ? 'uluna' : this._asset
```

Comment:
I thought that you can pay fee in any asset? Why not just respect whatever the user passed in constructor as `asset` ? 

If it's only possible to pay fee in luna, throw an error if `tokenAddress` is specified and asset is not luna. This avoids situation where a client passes an asset, and they think its going to be used